server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java

```diff
@@ -154,21 +151,17 @@ private void register(QProfileName name, Collection<RulesProfile> profiles, DbSe
   }
 
   private void setDefault(String language, List<RulesProfile> profileDefs, DbSession session) {
-    String propertyKey = "sonar.profile." + language;
     boolean upToDate = false;
-    String currentDefault = settings.getString(propertyKey);
+    QualityProfileDto currentDefault = dbClient.qualityProfileDao().getDefaultProfile(language, session);
     if (currentDefault != null) {
-      // check validity
-      QualityProfileDto profile = dbClient.qualityProfileDao().getByNameAndLanguage(currentDefault, language, session);
-      if (profile != null) {
-        upToDate = true;
-      }
+      upToDate = true;
     }
 
     if (!upToDate) {
```

Comment:
Method can be simplified. No need to use intermediary variable "upToDate". Just use condition "if (currentDefault == null)"